src/wrappers/any.jl

```diff
@@ -20,21 +20,54 @@ function __init__()
     @initcxx
 end  # __init__()
 
-function node_manager_port()
+function parse_ray_args()
+   #==
+    "Starting agent process with command: ...
+    --node-ip-address=127.0.0.1 --metrics-export-port=60404 --dashboard-agent-port=60493
+    --listen-port=52365 --node-manager-port=58888
+    --object-store-name=/tmp/ray/session_2023-08-14_14-54-36_055139_41385/sockets/plasma_store
+    --raylet-name=/tmp/ray/session_2023-08-14_14-54-36_055139_41385/sockets/raylet
+    --temp-dir=/tmp/ray --session-dir=/tmp/ray/session_2023-08-14_14-54-36_055139_41385
+    --runtime-env-dir=/tmp/ray/session_2023-08-14_14-54-36_055139_41385/runtime_resources
+    --log-dir=/tmp/ray/session_2023-08-14_14-54-36_055139_41385/logs
+    --logging-rotate-bytes=536870912 --logging-rotate-backup-count=5
+    --session-name=session_2023-08-14_14-54-36_055139_41385
+    --gcs-address=127.0.0.1:6379 --minimal --agent-id 470211272
+   ==#
     line = open("/tmp/ray/session_latest/logs/raylet.out") do io
         while !eof(io)
             line = readline(io)
-            if contains(line, "NodeManager server started")
+            if contains(line, "Starting agent process")
                 return line
             end
         end
     end
 
-    m = match(r"port (\d+)", line)
-    return m !== nothing ? parse(Int, m[1]) : error("Unable to find port")
+    gcs_match = match(r"gcs-address=(([0-9]{1,3}\.){3}[0-9]{1,3}:[0-9]{4})", line)
```

Comment:
how confident are we that the port will always be exactly 4 digits?

Reply:
It's currently hardcoded as "6379" but [ray let's it be user defined](https://docs.ray.io/en/latest/cluster/cli.html?highlight=6379#:~:text=the%20port%20of%20the%20head%20ray%20process.%20If%20not%20provided%2C%20defaults%20to%206379%3B%20if%20port%20is%20set%20to%200%2C%20we%20will%20allocate%20an%20available%20port.) so I should generalise it.